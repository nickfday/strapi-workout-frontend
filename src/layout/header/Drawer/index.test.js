import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import TemporaryDrawer from '.';
import storyData from './storiesData';

afterEach(cleanup);

describe('TemporaryDrawer', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TemporaryDrawer {...storyData} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<TemporaryDrawer {...storyData} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('toggles open/close correctly', () => {
    const { asFragment, debug } = render(<TemporaryDrawer {...storyData} />);
    debug();

    expect(document.querySelector('.MuiDrawer-paper')).toBe(null);

    fireEvent.click(document.querySelector('.menu-toggle'));

    expect(document.querySelector('.MuiDrawer-paper')).toBeTruthy;

    fireEvent.keyDown(document.body, { key: 'Tab', code: '9' });
    fireEvent.keyDown(document.body, { key: 'Shift', code: 'ShiftLeft' });

    expect(document.querySelector('.MuiDrawer-paper')).toBeTruthy;

    fireEvent.keyDown(document.body, { key: 'Escape', code: '27' });

    debug();

    // fireEvent.keyDown(document.querySelector('.MuiBackdrop-root'), { key: 'Escape', code: '27' });

    // fireEvent.click(document.querySelector('.MuiBackdrop-root'));

    // expect(document.querySelector('.MuiDrawer-paperAnchorLeft')).toBe(null);

    // expect(document.querySelector('a').href).toBe('http://localhost/exercises');
    // fireEvent.click(document.querySelector("button"));
    // expect(asFragment('getByLabelText')).toBe('menu');
  });
});
