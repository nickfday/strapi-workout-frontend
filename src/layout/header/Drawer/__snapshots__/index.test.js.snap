// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemporaryDrawer matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Drawer__StyledHeadingContainer-k3c2z5-1 budDzF"
    >
      <div
        class="MuiButtonGroup-root"
        role="group"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined menu-toggle"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          />
        </button>
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root Drawer__StyledMenuIcon-k3c2z5-0 QTLbc"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
